doc/source/serve/doc_code/model_composition/chaining_example.py

```diff
@@ -0,0 +1,54 @@
+# flake8: noqa
+# __chaining_example_start__
+# File name: chain.py
+from ray import serve
+from ray.serve.handle import DeploymentHandle, DeploymentResponse
+
+
+@serve.deployment
+class Adder:
+    def __init__(self, increment: int):
+        self._increment = increment
+
+    def __call__(self, val: int) -> int:
+        return val + self._increment
+
+
+@serve.deployment
+class Multiplier:
+    def __init__(self, multiple: int):
+        self._multiple = multiple
+
+    def __call__(self, val: int) -> int:
+        return val * self._multiple
+
+
+@serve.deployment
+class Ingress:
+    def __init__(self, adder, multiplier):
+        self._adder: DeploymentHandle = adder.options(
+            use_new_handle_api=True,
+        )
+        self._multiplier: DeploymentHandle = multiplier.options(
+            use_new_handle_api=True,
+        )
+
+    async def __call__(self, input: int) -> int:
+        adder_response: DeploymentResponse = self._adder.remote(input)
+        # Pass the adder response directly into the multipler (no `await` needed).
+        multiplier_response: DeploymentResponse = self._multiplier.remote(
```

Comment:
Would it be better to show an example where multiplier take a ref and can do some pre-processing before needing the output from the previous deployment call?

Reply:
added